packages/docusaurus-theme-classic/src/theme/DocItem/index.js

```diff
@@ -14,7 +14,75 @@ import DocPaginator from '@theme/DocPaginator';
 
 import styles from './styles.module.css';
 
+let headersCache = [];
+let tableOfContentLinksCache = [];
+let lastActiveTableOfContentLink;
+
+function getActiveHeader() {
+  const TOP_OFFSET = 80;
+  let index = 0;
+  let activeHeader;
+  while (index < headersCache.length && !activeHeader) {
+    const header = headersCache[index];
+    const {top} = header.getBoundingClientRect();
+    if (top > 0 && top <= TOP_OFFSET) {
+      activeHeader = header;
+    }
+    index += 1;
+  }
+  return activeHeader;
+}
+
+function highlightTableOfContentLink(tableOfContentLink) {
+  const cssClass = 'contents__link--active';
+  if (lastActiveTableOfContentLink) {
+    lastActiveTableOfContentLink.classList.remove(cssClass);
+  }
+  tableOfContentLink.classList.add(cssClass);
+  lastActiveTableOfContentLink = tableOfContentLink;
+}
+
+function setActiveTableOfContentLink() {
+  const activeHeader = getActiveHeader();
+  if (activeHeader) {
+    let index = 0;
+    let itemHighlighted = false;
+    while (index < tableOfContentLinksCache.length && !itemHighlighted) {
+      const tableOfContentLink = tableOfContentLinksCache[index];
+      if (activeHeader.innerText.indexOf(tableOfContentLink.innerText) > -1) {
```

Comment:
This is not a robust approach. We should be doing exact matches for `href`s instead of looking at the text. The `href` is the id of each header, and are guaranteed by our table of contents generation algorithm to be unique on the page.

For example, for https://deploy-preview-1896--docusaurus-2.netlify.com/docs/contributing#our-development-process

```
// Rendered Markdown
<h2>
  <a aria-hidden="true" class="anchor" id="our-development-process"></a>
  <a aria-hidden="true" class="hash-link" href="#our-development-process">#</a>Our development process
</h2>

// Table of Contents
<a href="#our-development-process" class="contents__link">Our development process</a>
```

Reply:
Changed, is going to be up in the next commit.